sunspot/lib/sunspot/query/sort_composite.rb

```diff
@@ -19,12 +19,29 @@ def <<(sort)
       end
 
       # 
-      # Combine the sorts into a single param by joining them
+      # Combine the sorts into a single sort-param by joining them and add
+      # possible custom additional params
       #
       def to_params(prefix = "")
         unless @sorts.empty?
           key = "#{prefix}sort".to_sym
-          { key => @sorts.map { |sort| sort.to_param } * ', ' }
+          params = { key => @sorts.map { |sort| sort.to_params[:sort] } * ', ' }
+          @sorts.each do |sort|
+            sort.to_params.each do |param, value|
+              next if param == :sort
+              param = param.to_sym
+              if params.has_key?(param) && params[param] != value
```

Comment:
We may want to consider renaming some things here - `params[param]` is fairly confusing.